### PR TITLE
feat: add mirror mode support

### PR DESCRIPTION
**Pull request by guoyingtao**

### CODE REVIEW

File: Sources/Mantis/CropView/CropViewModel.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 import UIKit
 
-enum ImageRotationType: CGFloat {
+public enum ImageRotationType: CGFloat {
```
Comment (vanniktech) [inline Sources/Mantis/CropView/CropViewModel.swift:11]: Why do we need this to be public now?

File: Sources/Mantis/CropViewController/CropToolbar.swift [hunk 2]
```diff
@@ -40,6 +40,20 @@ public class CropToolbar: UIView, CropToolbarProtocol {
         button.setImage(icon, for: .normal)
         return button
     }()
+    
+    private lazy var horizontallyFlipButton: UIButton = {
+        let button = createOptionButton(withTitle: nil, andAction: #selector(horizontallyFlip))
+        let icon = iconProvider?.getClockwiseRotationIcon() ?? ToolBarButtonImageBuilder.horizontallyFlipImage()
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropToolbar.swift:46]: `iconProvider?.getClockwiseRotationIcon()` seems like a copy paste mistake. 

```suggestion
        let icon = iconProvider?.getHorizontallyFlipIcon() ?? ToolBarButtonImageBuilder.horizontallyFlipImage()
```

File: Sources/Mantis/CropViewController/CropToolbar.swift [hunk 3]
```diff
@@ -40,6 +40,20 @@ public class CropToolbar: UIView, CropToolbarProtocol {
...
+        button.setImage(icon, for: .normal)
+        return button
+    }()
+    
+    private lazy var verticallyFlipButton: UIButton = {
+        let button = createOptionButton(withTitle: nil, andAction: #selector(verticallyFlip(_:)))
+        let icon = iconProvider?.getClockwiseRotationIcon() ?? ToolBarButtonImageBuilder.verticallyFlipImage()
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropToolbar.swift:53]:
```suggestion
        let icon = iconProvider?.getVerticallyFlipIcon() ?? ToolBarButtonImageBuilder.verticallyFlipImage()
```

File: Sources/Mantis/CropViewController/CropToolbar.swift [hunk 4]
```diff
@@ -203,6 +225,14 @@ extension CropToolbar {
     @objc private func alterCropper90Degree(_ sender: Any) {
         cropToolbarDelegate?.didSelectAlterCropper90Degree()
     }
+    
+    @objc private func horizontallyFlip(_ sender: Any) {
+        cropToolbarDelegate?.didHorizontallyFlip()
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropToolbar.swift:230]:
```suggestion
        cropToolbarDelegate?.didSelectHorizontallyFlip()
```

File: Sources/Mantis/CropViewController/CropToolbar.swift [hunk 5]
```diff
@@ -203,6 +225,14 @@ extension CropToolbar {
...
+    }
+
+    @objc private func verticallyFlip(_ sender: Any) {
+        cropToolbarDelegate?.didVerticallyFlip()
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropToolbar.swift:234]:
```suggestion
        cropToolbarDelegate?.didSelectVerticallyFlip()
```

File: Sources/Mantis/CropViewController/CropToolbarProtocol.swift [hunk 6]
```diff
@@ -16,6 +19,8 @@ public protocol CropToolbarDelegate: AnyObject {
     func didSelectSetRatio()
     func didSelectRatio(ratio: Double)
     func didSelectAlterCropper90Degree()
+    func didHorizontallyFlip()
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropToolbarProtocol.swift:22]: For consistency:

```suggestion
    func didSelectHorizontallyFlip()
```

File: Sources/Mantis/CropViewController/CropToolbarProtocol.swift [hunk 7]
```diff
@@ -16,6 +19,8 @@ public protocol CropToolbarDelegate: AnyObject {
...
+    func didVerticallyFlip()
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropToolbarProtocol.swift:23]:
```suggestion
    func didSelectVerticallyFlip()
```

File: Sources/Mantis/CropViewController/CropViewController.swift [hunk 8]
```diff
@@ -432,6 +432,14 @@ public class CropViewController: UIViewController {
         }
     }
     
+    private func handleHorizontallyFlp() {
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropViewController.swift:435]:
```suggestion
    private func handleHorizontallyFlip() {
```

File: Sources/Mantis/CropViewController/CropViewController.swift [hunk 9]
```diff
@@ -556,6 +564,14 @@ extension CropViewController: CropToolbarDelegate {
     public func didSelectAlterCropper90Degree() {
         handleAlterCropper90Degree()
     }
+    
+    public func didHorizontallyFlip() {
+        handleHorizontallyFlp()
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/CropViewController.swift:569]:
```suggestion
        handleHorizontallyFlip()
```

File: Sources/Mantis/CropViewController/ToolBarButtonImageBuilder.swift [hunk 10]
```diff
@@ -76,4 +76,36 @@ struct ToolBarButtonImageBuilder {
...
+        if #available(macCatalyst 13.1, iOS 13.0, *) {
+            return UIImage(systemName: "flip.horizontal")
+        }
+        
+        return nil
+    }
+    
+    static func verticallyFlipImage() -> UIImage? {
+        if #available(macCatalyst 13.1, iOS 13.0, *) {
+            guard let horizontallyFippedImage = horizontallyFlipImage(),
```
Comment (vanniktech) [inline Sources/Mantis/CropViewController/ToolBarButtonImageBuilder.swift:90]: Nice idea that you just take the other icon and rotate / translate it!

**Review by vanniktech — CHANGES_REQUESTED** [review]
Works really well for me!

File: Sources/Mantis/CropView/CropViewModel.swift (hunk 1 above)
Comment (guoyingtao) [reply Sources/Mantis/CropView/CropViewModel.swift:11]: I think I forgot deleting some test code. I will remove `public`

File: Sources/Mantis/CropViewController/CropToolbarProtocol.swift (hunk 6 above)
Comment (guoyingtao) [reply Sources/Mantis/CropViewController/CropToolbarProtocol.swift:22]: Thanks for these renaming suggestions!

File: Sources/Mantis/CropView/CropViewModel.swift (hunk 1 above)
Comment (guoyingtao) [reply Sources/Mantis/CropView/CropViewModel.swift]: Fixed

**Review by vanniktech — APPROVED** [review]
Amazing! Thank you for the implementation ❤️

**Pull request merged**